### PR TITLE
Remove terraform string matching

### PR DESCRIPTION
**Pull request by alexdavid**
Use hashicorp's hcl package to test our outputted terraform modules without doing raw string comparisons

### CODE REVIEW

**Comment by alexdavid** [comment]
Passing tests blocking on #774

**Review by kevgo — COMMENTED** [review]
This is really cool. Nice use of custom-built matchers that lean on the HCL parser to parse and verify HCL output!

File: src/terraform/index_test.go [hunk 1]
```diff
@@ -39,44 +40,22 @@ var _ = Describe("Template builder", func() {
...
+				"key_name":          "${var.key_name}",
+				"name":              "example-app",
+				"env":               "production",
+				"external_dns_name": "example-app.com",
+			}))
 		})
 	})
 
 	var _ = Describe("Given an application with public and worker services", func() {
 		var result string
```
Comment (charlierudolph) [inline src/terraform/index_test.go]: I don't think we need this variable anymore

File: src/terraform/index_test.go [hunk 2]
```diff
@@ -119,68 +98,56 @@ module "aws" {
...
-  region                = "${module.aws.region}"
-  ssl_certificate_arn   = "sslcert123"
-  vpc_id                = "${module.aws.vpc_id}"
-}`)
-			Expect(result).To(ContainSubstring(expected))
+			Expect(hclFile).To(matchers.HaveHCLVariable("public-service_env_vars"))
+			Expect(hclFile).To(matchers.HaveHCLVariable("public-service_docker_image"))
+			Expect(hclFile.Module["public-service"]).To(Equal(hcl.Module{
+				"source": "git@github.com:Originate/exosphere.git//terraform//aws//public-service?ref=TERRAFORM_MODULES_REF",
+
```
Comment (charlierudolph) [inline src/terraform/index_test.go]: why the empty line here? Same for below. I think it can be deleted

**Review by charlierudolph — APPROVED** [review]
Good to merge on green

**Pull request merged**